rasterio/io.py

```diff
@@ -221,7 +221,9 @@ def open(self, driver=None, sharing=False, **kwargs):
             raise IOError("I/O operation on closed file.")
         # Assume we were given a non-empty file-like object
         log.debug("VSI path: {}".format(mempath.path))
-        return DatasetReader(mempath, driver=driver, sharing=sharing, **kwargs)
+
+        with Env(GDAL_PAM_ENABLED=False):
+            return DatasetReader(mempath, driver=driver, sharing=sharing, **kwargs)
```

Comment:
as mentioned in https://github.com/rasterio/rasterio/discussions/2825#discussioncomment-5875592 I'm not quite sure about this. Why the behaviour should be different when using `FilePathBase` than with other method? 

It has always been up to the user to set GDAL environment 🤷‍♂️

I may totally misunderstood the issue here

Reply:
oh I think I understand, because a FilePath has not `path` it's not possible for GDAL to find any `linked` aux file because a FilePath can't have aux files anyway 😅

Reply:
@vincentsarago `GDAL_PAM_ENABLED=True`, which is the default, can never succeed with FilePath and only results in noise.